modules/command-engine/core/src/main/scala/surge/core/commondsl/SurgeCommandBusinessLogic.scala

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package surge.core.commondsl
+
+import com.typesafe.config.Config
+import io.opentracing.Tracer
+import io.opentracing.noop.NoopTracerFactory
+import surge.core.command.{ AggregateCommandModelCoreTrait, SurgeCommandKafkaConfig }
+import surge.core.{ SurgeAggregateReadFormatting, SurgeWriteFormatting }
+import surge.kafka.KafkaTopic
+import surge.metrics.Metrics
+
+trait SurgeGenericBusinessLogicTrait[AggId, Agg, Command, Rej, Event] {
```

Comment:
Wondering if there's any usefulness in a common base trait between `AbstractSurgeEventBusinessLogic` and `SurgeGenericBusinessLogicTrait`.  They're really similar except that the event business logic takes in an event model, and the command business logic a command model and additionally an events topic.  Otherwise they're roughly the same.

It may be nice for the two to evolve independently I suppose, but it does look really similar at the moment.

Reply:
Good suggestion.  I think this commit goes a long way to cleaning this up.